File: Examples/ExceptionRecovery/ExceptionRecovery.java
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.0

public final class ExceptionRecovery {
    private ExceptionRecovery() {}

    private static int requirePositive(int value) {
        if (value <= 0) {
            throw new IllegalArgumentException("value must be positive");
        }
        return value;
    }

    public static void main(String[] args) {
        int[] inputs = {3, 0, 7};
        for (var input : inputs) {
            try {
                System.out.println(requirePositive(input));
            } catch (IllegalArgumentException error) {
                System.err.println(error.getMessage());
            }
        }
    }
}

File: Examples/ExceptionRecovery/ExceptionRecovery.cs
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.1

using System;

namespace Examples.ExceptionRecovery;

public static class ExceptionRecovery
{
    private static int RequirePositive(int value) => value > 0
        ? value
        : throw new ArgumentOutOfRangeException(nameof(value), "value must be positive");

    public static void Main()
    {
        int[] inputs = [3, 0, 7];
        foreach (var input in inputs)
        {
            try
            {
                Console.WriteLine(RequirePositive(input));
            }
            catch (ArgumentOutOfRangeException error)
            {
                Console.Error.WriteLine(error.Message);
            }
        }
    }
}
